### PR TITLE
Fix: Inline editing not working when the Optimized DOM experiment is on [ED-3081]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: includes/widgets/text-editor.php [hunk 1]
```diff
@@ -408,21 +408,22 @@ protected function register_controls() {
...
 	 */
 	protected function render() {
-		$editor_content = $this->get_settings_for_display( 'editor' );
+		$is_edit_mode = Plugin::$instance->editor->is_edit_mode();
 
+		$editor_content = $this->get_settings_for_display( 'editor' );
 		$editor_content = $this->parse_text_editor( $editor_content );
 
-		if ( ! Plugin::$instance->experiments->is_feature_active( 'e_dom_optimization' ) ) {
+		if ( $is_edit_mode ) {
```
Comment (matipojo) [inline includes/widgets/text-editor.php:416]: Where do you check if the Experiment is active? (PHP & JS)

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/widgets/text-editor.php (hunk 1 above)
Comment (StyleShit) [reply includes/widgets/text-editor.php:416]: I don't.
After thinking about it, I've realized that it doesn't matter, we want this div only in edit mode, and we don't care about the DOM experiment.

Comment (matipojo) [reply includes/widgets/text-editor.php:416]: I can't revert it, please roll it back. it's for BC.

Comment (StyleShit) [reply includes/widgets/text-editor.php:416]: Why? Looks like it should work for BC anyway.

Comment (matipojo) [reply includes/widgets/text-editor.php:416]: If the experiment is inactive the div should be printed.

Comment (StyleShit) [reply includes/widgets/text-editor.php]: Done.

**Review by matipojo — APPROVED** [review]

**Pull request merged**